monkey/tests/unit_tests/monkey_island/cc/agent_event_handlers/test_update_machine_os.py

```diff
@@ -0,0 +1,85 @@
+from ipaddress import IPv4Interface
+from unittest.mock import MagicMock
+from uuid import UUID
+
+import pytest
+
+from common import OperatingSystem
+from common.agent_events import OSDiscoveryEvent
+from common.types import SocketAddress
+from monkey_island.cc.agent_event_handlers import update_machine_os
+from monkey_island.cc.models import Agent, Machine
+from monkey_island.cc.repository import (
+    AgentMachineFacade,
+    IAgentRepository,
+    IMachineRepository,
+    UnknownRecordError,
+)
+
+# The machine
+MACHINE_ID = 99
+MACHINE = Machine(
+    id=MACHINE_ID, hardware_id=33, network_interfaces=[IPv4Interface("10.10.10.55/24")]
+)
+
+# The agent
+AGENT_ID = UUID("72a64013-b3ab-4be9-9f05-0ffaccf01950")
+CC_SERVER = SocketAddress(ip="10.10.10.100", port="5000")
+AGENT = Agent(id=AGENT_ID, machine_id=MACHINE_ID, start_time=0, cc_server=CC_SERVER)
+
+# The event
+EVENT = OSDiscoveryEvent(source=AGENT_ID, os=OperatingSystem.LINUX, version="blah")
+
+
+@pytest.fixture
+def agent_repository() -> IAgentRepository:
+    repository = MagicMock(spec=IAgentRepository)
+    repository.get_agent_by_id = MagicMock(return_value=AGENT)
+    return repository
+
+
+MACHINES_BY_ID = {MACHINE_ID: MACHINE}
+
+
+def machine_from_id(id):
+    return MACHINES_BY_ID[id]
+
+
+@pytest.fixture
+def machine_repository() -> IMachineRepository:
+    repository = MagicMock(spec=IMachineRepository)
+    repository.get_machine_by_id = MagicMock(side_effect=machine_from_id)
+    return repository
+
+
+@pytest.fixture
+def agent_machine_facade(agent_repository, machine_repository) -> AgentMachineFacade:
+    return AgentMachineFacade(agent_repository, machine_repository)
+
+
+@pytest.fixture
+def updater(agent_machine_facade, machine_repository) -> update_machine_os:
+    return update_machine_os(agent_machine_facade, machine_repository)
+
+
+def test_machine_updated(updater, machine_repository):
+    updater(EVENT)
+    assert machine_repository.upsert_machine.called
```

Comment:
If you use `InMemoryMachineRepository`, you can verify that the repository contents are correct.